src/Tests/Microsoft.Diagnostics.Monitoring.HostingStartup.UnitTests/ParameterCapturing/Pipeline/ParameterCapturingPipelineTests.cs

```diff
@@ -341,6 +330,51 @@ public async Task Request_StopsAfterDuration()
             Assert.Equal(payload.RequestId, stoppedRequest);
         }
 
+        [Fact]
+        public async Task Request_StopsAfterCaptureLimit()
```

Comment:
Consider parameterizing this test with a capture limit parameter. I'd like for it to test "the corners" (e.g. capture limit 0, 1, 2) to make sure no assumptions are made about the limit.